### PR TITLE
May 8th Python patchset: py3.11 + DISTUTILS_GLOBAL_ARGS in PEP517 + buildsys in logs + maybe sipbuild

### PR DESCRIPTION
**Pull request by mgorny**
Planned features:
- [x] sipbuild support (presuming new upstream git works)

### CODE REVIEW

**Comment by mgorny** [comment]
CC @gentoo/python

**Comment by mgorny** [comment]
CC @ionenwks for sipbuild stuff
CC @gentoo/qt for qmake-utils changes

**Review by arthurzam — APPROVED** [review]

**Review by thesamesam — APPROVED** [review]
Can you ensure the wiki (or python guides porting section) includes all these steps wrt new impls?

**Comment by mgorny** [comment]
> Can you ensure the wiki (or python guides porting section) includes all these steps wrt new impls?

Yes, I'll update the PG shortly. I'm also going to update the section on initial testing as I go with py3.11, as the previous approach isn't going to work anymore.

File: eclass/qmake-utils.eclass [hunk 1]
```diff
@@ -60,6 +60,33 @@ qt5_get_plugindir() {
...
+		QMAKE_STRIP=
+		QMAKE_CFLAGS="${CFLAGS}"
+		QMAKE_CFLAGS_RELEASE=
+		QMAKE_CFLAGS_DEBUG=
+		QMAKE_CXXFLAGS="${CXXFLAGS}"
+		QMAKE_CXXFLAGS_RELEASE=
+		QMAKE_CXXFLAGS_DEBUG=
+		QMAKE_LFLAGS="${LDFLAGS}"
+		QMAKE_LFLAGS_RELEASE=
+		QMAKE_LFLAGS_DEBUG=
```
Comment (Pesa) [inline eclass/qmake-utils.eclass:86]: (minor) can we avoid duplicating these options both here and in `eqmake5`? these rarely/never change, so I'm fine either way.

**Review by Pesa — APPROVED** [review]
ACK for qmake-utils and sip

File: eclass/qmake-utils.eclass (hunk 1 above)
Comment (mgorny) [reply eclass/qmake-utils.eclass:86]: I'll try to figure something sane out.

Comment (mgorny) [reply eclass/qmake-utils.eclass:86]: Please check the latest commit out and tell me if it's to your liking.

File: eclass/qmake-utils.eclass [hunk 2]
```diff
@@ -102,28 +102,10 @@ eqmake5() {
...
-		QMAKE_CFLAGS_DEBUG= \
-		QMAKE_CXXFLAGS="${CXXFLAGS}" \
-		QMAKE_CXXFLAGS_RELEASE= \
-		QMAKE_CXXFLAGS_DEBUG= \
-		QMAKE_LFLAGS="${LDFLAGS}" \
-		QMAKE_LFLAGS_RELEASE= \
-		QMAKE_LFLAGS_DEBUG= \
-		"$@"
+	local -a args
+	mapfile -t args <<<"$(qt5_get_qmake_args)"
```
Comment (Pesa) [inline eclass/qmake-utils.eclass:106]: hmm can we use `mapfile -t`? I think it was added in bash-5.0 or so and we still need to support 4.2 for EAPIs < 8.

Comment (ionenwks) [reply eclass/qmake-utils.eclass:106]: mapfile was added in bash-4.0 and it support -t from the start, we're good

Comment (Pesa) [reply eclass/qmake-utils.eclass:106]: great, then +1 from me, thanks!

Comment (mgorny) [reply eclass/qmake-utils.eclass:106]: Thanks!

**Review by Pesa — APPROVED** [review]

**Pull request closed without merging**